### ISSUE TITLE
Support for webflux endpoints

### ISSUE THREAD

**Issue by dsyer** [spring-cloud/spring-cloud-config#885]
There's no real benefit in using webflux, but it might be possible to embed the config server in a webflux app, in principle at least.

**Comment by dharezlak** [comment]
This is exactly our use case at the moment. We would like to have a config server which exposes a small REST API (webflux) to modify the provided configuration (JDBC-based). The API will be used by a web configuration panel so :+1:

**Comment by spencergibb** [comment]
You know there's no official spring supported non-blocking relational db access yet (jdbc will never be non-blocking)? Things are in the works though.

**Comment by dsyer** [comment]
@venkatnpedada that doesn't really make any sense to me. Why would you need a config server embedded in an integration test?

**Comment by venkatnpedada** [comment]
@dsyer In my case, I am pulling application properties directly from git server instead of a cloud config server. This will allow me to run integration test cases without depending on a config server to pull the application properties.

**Comment by wdyun** [comment]
So, how to use embed the config server with webflux? I have same problem.

**Comment by spencergibb** [comment]
Not currently possible

**Comment by sgornostal** [comment]
do you have any updates on this?

**Comment by spencergibb** [comment]
No, when there is, you will see assignments to projects and milestones.

**Comment by gdomo** [comment]
This is our use case at the moment, too. We ship webflux web-application with embed config server.

**Comment by krispanalov** [comment]
Is there a way to vote for this topic. In our case we want to have all our services written with webflux and so are our libs. Now we will need to support both of them.

**Comment by spencergibb** [comment]
Add a reaction to the very first comment

**Comment by dlicois** [comment]
It is currently impossible to embed a spring cloud config server in a gateway as long as it isn't resolved..

**Comment by jgcollings** [comment]
I would also like to embed config server in my webflux cloud gateway. I don't want to host a separate config server. When I attempt to include the spring-cloud-config-server dependency it adds webmvc to the class path and breaks everything. Adding my vote to fix this.
